web/crux/src/interceptors/grpc.user.interceptor.ts

```diff
@@ -0,0 +1,49 @@
+import { Metadata } from '@grpc/grpc-js'
+import { CallHandler, ExecutionContext, Injectable, Logger, NestInterceptor, SetMetadata } from '@nestjs/common'
+import { Reflector } from '@nestjs/core'
+import { Observable } from 'rxjs'
+import KratosService from 'src/services/kratos.service'
+
+const DISABLE_ACCESSED_BY_METADATA = 'disable-accessed-by-metadata'
+
+export const DisableAccessedByMetadata = () => SetMetadata(DISABLE_ACCESSED_BY_METADATA, true)
+
+export const setAccessedByFromCookie = async (metadata: Metadata, kratos: KratosService) => {
+  const cookie = metadata.getMap().cookie as string
+  if (!cookie) {
+    throw new Error('Cookie missing')
+  }
+
+  const session = await kratos.getSessionByCookie(cookie)
+
+  metadata.add('accessedby', session.identity.id)
```

Comment:
`accessedby` withouth camelCase is okey? Otherhand use const as DISABLED_ACCESSED... would be better, IMHO.

Reply:
For some reason metadata keys are lowercased by default, so even if I used `accessedBy` when trying to get the key it only works with `accessedby`.

Putting it into a const is a good idea though 👍